crates/hir-def/src/resolver.rs

```diff
@@ -247,6 +246,24 @@ impl Resolver {
                 }
                 Scope::BlockScope(m) => {
                     if let Some(res) = m.resolve_path_in_type_ns(db, path) {
+                        let res = match res.0 {
+                            TypeNs::ModuleId(_) => {
```

Comment:
```suggestion
                            TypeNs::ModuleId(_) if res.1.is_none() => {
```
Otherwise this is a partially resolved path to the module, then we wan't to continue resolution for the module base

Reply:
I see. If res is a `BuiltinType`, then the `segment_index` is None.